src/coreclr/jit/jitconfig.cpp

```diff
@@ -320,100 +92,90 @@ void JitConfigValues::MethodSet::destroy(ICorJitHost* host)
     m_names = nullptr;
 }
 
-// strstr that is length-limited, this implementation is not intended to be used on hot paths
-static size_t strnstr(const char* pSrc, size_t srcSize, const char* needle, size_t needleSize)
+static bool matchGlob(const char* pattern, const char* patternEnd, const char* str)
 {
-    for (size_t srcPos = 0; srcPos <= srcSize - needleSize; srcPos++)
+    while (true)
     {
-        if (strncmp(pSrc + srcPos, needle, needleSize) == 0)
+        if (pattern == patternEnd)
         {
-            return srcPos;
+            return *str == '\0';
         }
-    }
-    return -1;
-}
 
-static bool matchesName(
-    const char* const name, int nameLen, bool wildcardAtStart, bool wildcardAtEnd, const char* const s2)
-{
-    if (wildcardAtStart && (int)strnstr(s2, strlen(s2), name, nameLen) == -1)
-    {
-        return false;
-    }
+        if (*pattern == '*')
+        {
+            while (true)
+            {
+                if (matchGlob(pattern + 1, patternEnd, str))
+                    return true;
 
-    if (!wildcardAtStart && strncmp(name, s2, nameLen) != 0)
-    {
-        // 's2' must start with 'nameLen' characters of 'name'
-        return false;
-    }
+                if (*str == '\0')
+                    return false;
 
-    // if we don't have a wildcardAtEnd then s2 also need to be zero terminated
-    if (!wildcardAtEnd && (s2[nameLen] != '\0'))
-    {
-        return false;
-    }
+                str++;
+            }
 
-    // we have a successful match
-    return true;
+            return false;
+        }
+
+        if (*str == '\0')
+        {
+            return false;
+        }
+
+        if ((*pattern != '?') && (*pattern != *str))
+        {
+            return false;
+        }
+
+        pattern++;
+        str++;
+    }
 }
 
 bool JitConfigValues::MethodSet::contains(const char*       methodName,
                                           const char*       className,
                                           CORINFO_SIG_INFO* sigInfo) const
 {
-    int numArgs = sigInfo != nullptr ? sigInfo->numArgs : -1;
+    // names[hasClassName][hasSignature]
+    char*     names[2][2] = {};
+    Compiler* comp        = JitTls::GetCompiler();
```

Comment:
Are we worried about this? It should be a single call in release builds and only when `JitDisasm` is non-empty.
We need it to allocate memory in rare cases where the string we are matching against is longer than 1024 characters.